apps/courses/tests/test_admin_course.py

```diff
@@ -0,0 +1,157 @@
+
+        # Get the admin change view
+        url = reverse("admin:courses_course_add")
+        response = self.client.get(url, follow=True)
+
+        # Check that the page includes all our editable fields
+        for field in ["active_session", "main_organization"]:
+            self.assertContains(response, "id_{:s}".format(field))
+
+    def test_course_change_view_get(self):
+        """
+        The admin change view should include the editable and readonly fields as expected.
+        In particular, the relation fields should only include options for related objects in
+        their draft version.
+        """
+        user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=user.username, password="password")
+
+        # Create a course
+        course = CourseFactory()
+
+        # Create an organization and publish it
+        organization = OrganizationFactory()
+        organization.extended_object.publish("en")
+        organization.refresh_from_db()
+
+        # Create a subject and publish it
+        subject = SubjectFactory()
+        subject.extended_object.publish("en")
+        subject.refresh_from_db()
+
+        # Get the admin change view
+        url = reverse("admin:courses_course_change", args=[course.id])
+        response = self.client.get(url)
+
+        # Check that the page includes all our fields
+        self.assertContains(
+            response, course.extended_object.get_title(), status_code=200
+        )
+        self.assertContains(response, course.active_session)
+        self.assertContains(
+            response, course.main_organization.extended_object.get_title()
+        )
+        # Only the draft organization and subject should be proposed as options in select boxes
+        self.assertContains(
+            response, '<option value="{:d}">{!s}'.format(subject.id, subject)
+        )
+        self.assertNotContains(
+            response,
+            '<option value="{:d}">{!s}'.format(
+                subject.public_extension.id, subject.public_extension
+            ),
+        )
+        self.assertContains(
+            response, '<option value="{:d}">{!s}'.format(organization.id, organization)
+        )
+        self.assertNotContains(
+            response,
+            '<option value="{:d}">{!s}'.format(
+                organization.public_extension.id, organization.public_extension
+            ),
+        )
+
+    def test_course_change_view_post(self):
+        """
+        Validate that the course can be updated via the admin.
+
+        In particular, when a course is updated from the admin, the main organization should still
+        be added as a many-to-many relation in "organizations". This is being taken care of by
+        overriding the `save_related` method on the ModelAdmin and not by the override to the
+        `save` method on the model (see explanation in the docstring of the`save_related` method
```

Comment:
I think I get your point, but this phrase needs to be a bit rephrase.

Reply:
I think the problem is better explained in the docstring of the "save_related" method. 
So in this docstring, I propose to just say:
```python
In particular, make sure that when a course is updated from the admin, the main organization is
automatically added to the many-to-many field "organizations".
See http://stackoverflow.com/a/1925784/469575 for details on the issue.
```

Reply:
Yes.